include/nlohmann/detail/output/serializer.hpp

```diff
@@ -193,7 +193,10 @@ class serializer
                     return;
                 }
 
-                if (pretty_print)
+                // For multi-dim number arrays, print the inner most dimension in one line to improve readability.
```

Comment:
I think the comment is misleading: you also change the behavior for 1-dimensional arrays.